### PR TITLE
Re-Identification application

### LINKED ISSUE #58: Re-id app to merge bboxes into tracks after TF annotation
Hi, great tool.  For ground-truth annotation, there are often too many objects in every frame.  It would be tremendously tedious work to annotate the track for every single one of objects.  Is there any pre-trained model or a way to run a custom model that can detect possibly identical objects, and all I have to do is to review and merge their tracks/IDs into one?

thanks

### PR DESCRIPTION
**Pull request by bsekachev**
Resolved #58

### CODE REVIEW

**Comment by bsekachev** [comment]
@nmanovic 
UPD: Now codacy issues fixed

**Comment by nmanovic** [comment]
- Use engine.job.change permissions
- Need to check that we can use images "after.gif" and "before.gif". I will prefer to prepare screencast using an approved video clip.

**Comment by bsekachev** [comment]
@nmanovic 
Done.

File: cvat/apps/reid/static/reid/js/enginePlugin.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ * Copyright (C) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+/* global showMessage userConfirm */
+
+
+document.addEventListener('DOMContentLoaded', () => {
+    function run(reidButton, tresholdInput, distanceInput) {
```
Comment (nmanovic) [inline cvat/apps/reid/static/reid/js/enginePlugin.js]: I see a lot of typos: treshold --> threshold! (in UI as well)

File: cvat/apps/reid/static/reid/js/enginePlugin.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            $(`#${reidWindowId}`).removeClass('hidden');
+        }
+    }).addClass('menuButton semiBold h2').prependTo(buttonsUI);
+
+    $(`
+        <div class="modal hidden" id="${reidWindowId}">
+            <div class="modal-content" style="width: 300px; height: 170px;">
+                <table>
+                    <tr>
+                        <td> <label class="regular h2"> Treshold: </label> </td>
```
Comment (nmanovic) [inline cvat/apps/reid/static/reid/js/enginePlugin.js]: Need to add hints. Also specify that the radius in pixels.

**Comment by nmanovic** [comment]
@bsekachev , the PR is fine. Just need to add a couple of hints and fix a couple of typos.

File: cvat/apps/reid/reid.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+        return box_paths
+
+
+    def run(self):
+        box_paths = self.__apply_matching()
+        output = []
+
+        # ReID process has been canceled
+        if box_paths is None:
```
Comment (nmanovic) [inline cvat/apps/reid/reid.py]: Is the line necessary?

**Comment by nmanovic** [comment]
@bsekachev , Should we "freeze" ui windows when the algorithm works? All changes in annotation will be lost (at least for bounding boxes)

**Comment by bsekachev** [comment]
@nmanovic 
Yes, all changes during ReID is running will be lost.

File: cvat/apps/reid/static/reid/js/enginePlugin.js (hunk 2 above)
Comment (bsekachev) [reply cvat/apps/reid/static/reid/js/enginePlugin.js]: Done

File: cvat/apps/reid/static/reid/js/enginePlugin.js (hunk 1 above)
Comment (bsekachev) [reply cvat/apps/reid/static/reid/js/enginePlugin.js]: Done

**Pull request merged**